src/tox_travis/hooks.py

```diff
@@ -1,23 +1,28 @@
 """Tox hook implementations."""
 import tox
+from .utils import parser_cfg, config_cfg
 from .toxenv import default_toxenv, override_ignore_outcome
 from .after import travis_after_monkeypatch
 
 
-@tox.hookimpl
+@tox.hookimpl(trylast=True)
 def tox_addoption(parser):
     """Add arguments and override TOXENV."""
     parser.add_argument(
         '--travis-after', dest='travis_after', action='store_true',
         help='Exit successfully after all Travis jobs complete successfully.')
 
-    default_toxenv()
+    cfg = parser_cfg(parser)
+    if cfg:
+        default_toxenv(cfg)
 
 
 @tox.hookimpl
 def tox_configure(config):
     """Check for the presence of the added options."""
+    cfg = config_cfg(config)
+
     if config.option.travis_after:
-        travis_after_monkeypatch()
+        travis_after_monkeypatch(cfg)
 
-    override_ignore_outcome(config)
+    override_ignore_outcome(config, cfg)
```

Comment:
I agree that it's confusing. The competing goal which motivated using `cfg` instead of `config` is that 1) almost nothing needs the full config, and 2) `config._cfg` is obviously an implementation detail of `config`, and I'd rather not scatter unstable APIs everywhere. Still it's confusing, and I can concede that there's not enough places we have to work with it that it's too big a deal.